### PR TITLE
Figure.meca: Refactor the tests for pass a single focal mechanism to spec

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

The `Figure.meca` tests are a little messy and difficult to maintain because each test uses different focal mechanisms and event information. I plan to refactor the tests and this PR is the first step. 

This PR focuses on testing passing a single focal mechanism to the `spec` parameter using different data types (dict, file, 1darray and DataFrame). Note that all tests now single a single baseline image `test_meca_spec_dict.png`.

Changes in this PR:

- Rename the test test_meca_spec_dictionary to test_meca_spec_dict
- Improve the test_meca_spec_dict test with a better baseline image
- Rewrite test_meca_spec_1d_array test to use the existing baseline image
- Rewrite the test_meca_spec_file test to reuse the existing baseline image
- Rename the test test_meca_spec_dataframe to test_meca_spec_two_dataframe
- Add a new test test_meca_spec_dataframe to test passing a dataframe to meca

### CODE REVIEW

File: pygmt/tests/test_meca.py [hunk 1]
```diff
@@ -54,7 +118,7 @@ def test_meca_spec_dict_list():
 
 
 @pytest.mark.mpl_image_compare
-def test_meca_spec_dataframe():
+def test_meca_spec_two_dataframe():
```
Comment (willschlitzer) [inline pygmt/tests/test_meca.py:121]: Why is this being renamed? As far as I can tell there is just a single DataFrame in this function.

Comment (seisman) [reply pygmt/tests/test_meca.py:121]: I was planning to have two groups of tests:

1. In tests `test_meca_spec_*`, a single focal mechanism in different data types (dict, DataFrame, 1darray and plaintext file) is passed to the `spec` parameter
2. In tests `test_meca_spec_two_*`, two focal mechanisms in different data types (dict with list as keys, DataFrame, 2D list, 2D array and plaintext file) are passed to the `spec` parameter.

That's why I added the `_two` suffix to the tests, but I agree that's not a good name for these tests. Do you have better ideas?

File: pygmt/tests/test_meca.py [hunk 2]
```diff
@@ -54,7 +118,7 @@ def test_meca_spec_dict_list():
...
     """
     Test supplying a pandas.DataFrame containing focal mechanisms and locations
     to the spec parameter.
```
Comment (willschlitzer) [inline pygmt/tests/test_meca.py:124]:
```suggestion
    Test supplying two focal mechanisms and their locations by passing a pandas.DataFrame 
    to the spec parameter.
```

I think this makes it more clear why "two" is in the function name.

File: pygmt/tests/test_meca.py (hunk 1 above)
Comment (willschlitzer) [reply pygmt/tests/test_meca.py:121]: That makes sense. I think it will be hard to come up with a descriptive test name without making it too wordy. I think it might just be easier to update the docstring (I made my suggestion below) to make it clear that it's passing two focal mechanisms.

**Comment by seisman** [comment]
I'm closing the issue and will open another PR to reorganize the meca tests later.
